src/engine/profiler.cc

```diff
@@ -116,10 +111,13 @@ OprExecStat *Profiler::AddOprStat(int dev_type, uint32_t dev_id) {
       return NULL;
   }
 
-  DevStat& dev_stat = profile_stat[idx];
-  {
-    std::lock_guard<std::mutex> lock{dev_stat.m_};
-    dev_stat.opr_exec_stats.push_back(opr_stat);
+  std::shared_ptr<DevStat> pstat = profile_stat_;  // hold a ref count
```

Comment:
why do you need to make a copy? If Profiler is not destructed profile_stat_ shouldn't be either, right?

Reply:
In a perfect world where the static shutdown works as intended, it shouldn't happen.  But currently, it is a possible that this function could be called after or during Profiler destruvtion.